News um Meta-Description und Meta-Titel erweitern (SEO)

eS-IT opened the pull request:
This PR implements #87.

eS-IT committed:
ADD: #87 Feature: News um Meta-Description und Meta-Titel erweitern (SEO)

Toflar commented on news-bundle/src/Resources/contao/classes/News.php:
```diff
@@ -494,6 +494,37 @@ public function purgeOldFeeds()
+            return; // no news
+        }
+
+        $news = \NewsModel::findOneBy(array('alias=?', 'published=?'), array($this->Input->get('items'), 1));
```
You can use `\Input::get('items')` here again, or even better, store it in a variable on line 506 :)

Toflar commented on news-bundle/src/Resources/contao/languages/de/tl_news.xlf:
```diff
@@ -25,6 +25,22 @@
         <source>Here you can change the author of the news item.</source>
         <target>Hier können Sie den Autor des Beitrags ändern.</target>
       </trans-unit>
+      <trans-unit id="tl_news.metatitle.0">
```
You only need to add English, the rest is managed on Transifex :)

Toflar reviewed (commented):
Two minor comments, excellent work otherwise :)

eS-IT committed:
REMOVED: German texts from news-bundle/src/Resources/contao/languages/de/tl_news.xlf

eS-IT committed:
CHANGED: store items in variable

Toflar commented on news-bundle/src/Resources/contao/languages/de/tl_news.xlf:
```diff
@@ -327,4 +327,4 @@
       </trans-unit>
     </body>
   </file>
-</xliff>
```
This file does not need to be adjusted :) (you've deleted the final new line. Our coding style forces to always end a file with a new line :)

eS-IT committed:
Update tl_news.xlf

removed final new line

leofeyer committed:
Fix the coding style

leofeyer commented on news-bundle/src/Resources/contao/config/config.php:
```diff
@@ -46,6 +46,7 @@
 $GLOBALS['TL_HOOKS']['generatePage'][] = array('contao_news.listener.generate_page', 'onGeneratePage');
 $GLOBALS['TL_HOOKS']['generateXmlFiles'][] = array('News', 'generateFeeds');
 $GLOBALS['TL_HOOKS']['replaceInsertTags'][] = array('contao_news.listener.insert_tags', 'onReplaceInsertTags');
+$GLOBALS["TL_HOOKS"]['generatePage'][] = array('News', 'setMetaData');
```
We already have a listener for the `generatePage` hook three lines above. Wouldn't it make sense to re-use it?

leofeyer commented:
Here's what needs to be done IMHO:

- [ ] Re-use the existing listener
- [ ] Rename the fields from `metatitle` to `title` and from `metadescription` to `description`, so they match `tl_page.title` and `tl_page.description`
- [ ] Put the fields into a separate palette analogous to `tl_page`

eS-IT commented:
In `tl_page` heißt das Feld `pageTitle`, da das Feld `title` der " Seitenname" ist. Soll es in `tl_news` `title` oder `pageTitle` heißen?

eS-IT committed:
Re-use the existing listener
Rename the fields from metatitle to title and from metadescription to description, so they match tl_page.title and tl_page.description
Put the fields into a separate palette analogous to tl_page

eS-IT committed:
Merge branch 'fix-87' of github.com:eS-IT/contao into fix-87

eS-IT committed:
Fixed code style

ausi reviewed (changes requested):
IMO `GeneratePageListener::setMetaData()` is missing a check if we are actually on a news page. If I got a calendar event with the same alias as a news entry, wouldn’t this code set the title/description for the news on the event page then?

Shouldn’t the code for setting the title/description instead go into the reader module as we already set the page title there? https://github.com/contao/contao/blob/578044764b17db7cbec78c94d0beae589dc6dbd6/news-bundle/src/Resources/contao/modules/ModuleNewsReader.php#L105

eS-IT committed:
fixed fieldnames in news-bundle/src/EventListener/GeneratePageListener.php

leofeyer commented:
> Shouldn’t the code for setting the title/description instead go into the reader module as we already set the page title there?

@ausi You are absolutely right.

leofeyer commented:
> In `tl_page` heißt das Feld `pageTitle`, da das Feld `title` der " Seitenname" ist. Soll es in `tl_news` `title` oder `pageTitle` heißen?

Gute Frage. Ich denke `tl_news.pageTitle`, weil das zumindest konsistent wäre.

eS-IT committed:
renamed tl_news.title to tl_news.pageTitle

contaoacademy commented:
Vielen herzlichen Dank, dass Ihr das Feature so schnell implementiert.

eS-IT committed:
Moved code for setting the title/description into the reader module

leofeyer committed:
Fix the coding style

leofeyer commented:
Thank you @eS-IT.
